packages/@aws-cdk/aws-location-alpha/lib/route-calculator.ts

```diff
@@ -0,0 +1,158 @@
+   */
+  public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee: grantee,
+      actions: actions,
+      resourceArns: [this.routeCalculatorArn],
+    });
+  }
+
+  /**
+   * Grant the given identity permissions to access to a route calculator resource to calculate a route.
+   *
+   * https://docs.aws.amazon.com/location/latest/developerguide/security_iam_id-based-policy-examples.html#security_iam_id-based-policy-examples-calculate-route
+   */
+  public grantRead(grantee: iam.IGrantable): iam.Grant {
+    return this.grant(grantee,
+      'geo:CalculateRoute',
+    );
+  }
+}
+
+/**
+ * A Route Calculator
+ *
+ * @see https://docs.aws.amazon.com/location/latest/developerguide/places-concepts.html
+ */
+export class RouteCalculator extends RouteCalculatorBase {
+  /**
+   * Use an existing route calculator by name
+   */
+  public static fromRouteCalculatorName(scope: Construct, id: string, routeCalculatorName: string): IRouteCalculator {
+    const routeCalculatorArn = Stack.of(scope).formatArn({
+      service: 'geo',
+      resource: 'route-calculator',
+      resourceName: routeCalculatorName,
+    });
+
+    return RouteCalculator.fromRouteCalculatorArn(scope, id, routeCalculatorArn);
+  }
+
+  /**
+   * Use an existing route calculator by ARN
+   */
+  public static fromRouteCalculatorArn(scope: Construct, id: string, routeCalculatorArn: string): IRouteCalculator {
+    const parsedArn = Stack.of(scope).splitArn(routeCalculatorArn, ArnFormat.SLASH_RESOURCE_NAME);
+
+    if (!parsedArn.resourceName) {
+      throw new Error(`Route Calculator Arn ${routeCalculatorArn} does not have a resource name.`);
+    }
+
+    class Import extends RouteCalculatorBase {
+      public readonly routeCalculatorName = parsedArn.resourceName!;
+      public readonly routeCalculatorArn = routeCalculatorArn;
+    }
+
+    return new Import(scope, id, {
+      account: parsedArn.account,
+      region: parsedArn.region,
+    });
+  }
+
+  public readonly routeCalculatorName: string;
+
+  public readonly routeCalculatorArn: string;
+
+  /**
+   * The timestamp for when the route calculator resource was created in ISO 8601 forma
+   *
+   * @attribute
+   */
+  public readonly routeCalculatorCreateTime: string;
+
+  /**
+   * The timestamp for when the route calculator resource was last updated in ISO 8601 format
+   *
+   * @attribute
+   */
+  public readonly routeCalculatorUpdateTime: string;
+
+  constructor(scope: Construct, id: string, props: RouteCalculatorProps = {}) {
+    if (props.routeCalculatorName && !Token.isUnresolved(props.routeCalculatorName) && !/^[-.\w]{1,100}$/.test(props.routeCalculatorName)) {
+      throw new Error(`Invalid route calculator name. The route calculator name must be between 1 and 100 characters and contain only alphanumeric characters, hyphens, periods and underscores. Received: ${props.routeCalculatorName}`);
+    }
+
```

Comment:
Can you please add validation for `description` length ([docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-location-routecalculator.html#cfn-location-routecalculator-description))?